cylc/flow/network/subscriber.py

```diff
@@ -95,8 +98,10 @@ async def subscribe(self, msg_handler, *args, **kwargs):
             if self.stopping:
                 break
             try:
-                [topic, msg] = await self.socket.recv_multipart()
+                [topic, msg] = await self.socket.recv_multipart(
+                    flags=zmq.NOBLOCK)
             except zmq.error.ZMQError:
+                await asyncio.sleep(NO_RECEIVE_INTERVAL)
```

Comment:
Does setting the ZMQ timeout not work with the PUB-SUB model?

Reply:
Even if it did, this is a legitimate/documented approach:
https://pyzmq.readthedocs.io/en/latest/api/zmq.html#zmq.Socket.recv

So if there's a reason to shift to an alternative we can, but until then, this is good to go IMO.